haystack/nodes/other/shaper.py

```diff
@@ -404,7 +425,7 @@ def run(  # type: ignore
         results = {}
         for output_key, output_value in zip(self.outputs, output_values):
             invocation_context[output_key] = output_value
-            if output_key in ["query", "file_paths", "labels", "documents", "meta"]:
+            if self.publish_outputs and output_key in ["query", "file_paths", "labels", "documents", "meta", "answers"]:
```

Comment:
I agree that with the finer control given by `publish_outputs`, the restriction applied by the list is not necessary anymore. Let's remove it

Reply:
Ok, I'll do so.